xlators/storage/posix/src/posix-gfid-path.c

```diff
@@ -196,7 +196,8 @@ posix_get_gfid2path(xlator_t *this, inode_t *inode, const char *real_path,
             if (size == 0)
                 goto done;
         }
-        list = alloca(size);
+
+        list = GF_MALLOC(size, gf_posix_mt_char);
         if (!list) {
             *op_errno = errno;
             goto err;
```

Comment:
`ret` needs to be set to '-1'.

Reply:
I've just seen this is a backport. I'll accept this change, but could you send another patch fixing this ?

Reply:
I have send a patch https://github.com/gluster/glusterfs/pull/1837 in master release to resolve the same.